Documentation/guides/OneDotNetEmbeddedResources.md

```diff
@@ -0,0 +1,280 @@
+  repackaged into `__AndroidLibraryProjects__.zip`. You can only have
+  one `@(LibraryProjectZip)` per project.
+* No support for `@(AndroidResource)` or `@(AndroidAsset)`.
+* `@(EmbeddedJar)` and `@(EmbeddedReferenceJar)` are packaged directly
+  as `EmbeddedResource` files.
+
+## New Approach in .NET 6
+
+We want to consider an approach that is NuGet-centric. When the above
+implementation was written, NuGet was just getting started. Developers
+still had the pattern of copying .NET assemblies from `bin` and
+committing it to their source control of choice. The single file
+approach worked well for this scenario.
+
+If we look at the general structure of a [NuGet package][nuget]:
+
+    lib
+        net6.0-android29\Foo.dll
+        net6.0-android30\Foo.dll
+    # Optional reference assemblies
+    ref
+        net6.0-android29\Foo.dll
+        net6.0-android30\Foo.dll
+    # Optional native libraries
+    runtimes
+        android-arm\native\libFoo.so
+        android-arm64\native\libFoo.so
+        android-x86\native\libFoo.so
+        android-x64\native\libFoo.so
+
+There is not a great place where all Android file types would fit
+following this pattern.
+
+In Android Studio, Android libraries are packaged as [`.aar`][aar]
+files. A Xamarin.Android library, `Foo.csproj`, could also generate an
+[`.aar`][aar] file in its `$(OutputPath)`:
+
+    Foo.aar
+        classes.jar
+        res/
+        assets/
+        libs/*.jar
+        jni/[arch]/*.so
+
+`@(AndroidEnvironment)` files are specific to Xamarin.Android, so we
+can make one addition to the file format:
+
+    Foo.aar
+        env/MyEnvironment.txt
+
+So the output of `Foo.csproj` would look like:
+
+    bin/Debug/[targetframework]/
+        Foo.dll
+        Foo.aar
+    bin/Release/[targetframework]/
+        Foo.dll
+        Foo.aar
+
+If you ran the `Pack` target, you would get a `.nupkg` file with:
+
+    lib
+        net6.0-android29\Foo.dll
+        net6.0-android29\Foo.aar
+        net6.0-android30\Foo.dll
+        net6.0-android30\Foo.aar
+
+When consuming the `.nupkg` files, Xamarin.Android will still have to
+extract [`.aar`][aar] files on disk for command-line tooling such as
+`javac`, `d8/r8` and `manifestmerger`. If users want to copy around
+loose build output and consume it, there will only be 1 additional
+[`.aar`][aar] file they will need to copy.
+
+### Java/Kotlin Dependencies
+
+A `Foo.csproj` might include `bar.aar` and `baz.jar` files that are
+Java/Kotlin dependencies.
+
+`.aar` files should sit alongside the .NET assembly:
+
+    lib
+        net6.0-android29\Foo.dll
+        net6.0-android29\Foo.aar
+        net6.0-android29\bar.aar
+        net6.0-android30\Foo.dll
+        net6.0-android30\Foo.aar
+        net6.0-android30\bar.aar
+
+The `baz.jar` file would be included in the above `Foo.aar` file at:
+
+    Foo.aar
+        classes.jar
+        libs/baz.jar
+
+### Native Libraries
+
+Since both `.aar` and `.nupkg` files support native libraries,
```

Comment:
What happens in the result of "collision", e.g. a referenced `.aar` and `.nupkg` *both* provide `libConflictingName.so`? Which "wins"?  (Should this even be specified? We could leave it explicitly unspecified, perhaps?)

Reply:
I think we should emit a build error on duplicates, I don't think we do this at all now. I added a comment on this.

Whichever native library was added to the `.apk` _last_ appears like it would win.

Reply:
I *don't* think this should be a build error.  "One can imagine" versioning scenarios in which NuGet A requires Nv1, while NuGet B requires Nv2, and -- through chicanery? -- you're able to get a final `.apk` which includes native libs from Nv2, and -- through testing -- know that it doesn't break NuGet A.

If this were an error, you couldn't even *build* such a thing in order to test it.

Do we want to *actively* prevent such a thing?  ¯\_(ツ)_/¯

Reply:
I think we should leave it as it would be today: you could have multiple `.dll` files with the same `@(EmbeddedNativeLibrary)` and we don't emit errors for this.

Whichever one happens to make it to the `.apk` is what you get.